centaur/src/main/scala/centaur/test/metadata/WorkflowFlatMetadata.scala

```diff
@@ -24,14 +24,18 @@ import scala.util.{Failure, Success, Try}
   */
 case class WorkflowFlatMetadata(value: Map[String, JsValue]) extends AnyVal {
 
-  def diff(actual: WorkflowFlatMetadata, workflowID: UUID, cacheHitUUID: Option[UUID] = None): Iterable[String] = {
+  def diff(actual: WorkflowFlatMetadata, workflowID: UUID, cacheHitUUID: Option[UUID] = None, compareTyped: Boolean = false): Iterable[String] = {
     // If the test fails in initialization there wouldn't be workflow root metadata, and if that's the expectation
     // then that's ok.
     val workflowRoot = actual.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
     val missingErrors = value.keySet.diff(actual.value.keySet) map { k => s"Missing key: $k" }
-    val mismatchErrors = value.keySet.intersect(actual.value.keySet) flatMap { k => diffValues(k, value(k), actual.value(k),
-      workflowID, workflowRoot, cacheHitUUID)}
-
+    val mismatchErrors = value.keySet.intersect(actual.value.keySet) flatMap { k =>
+      if (compareTyped) {
```

Comment:
[TOL] Personal preference, I find creating separate functions easier to follow than introducing a highly significant flag in an existing one
```
def diff(actual: WorkflowFlatMetadata, workflowID: UUID, cacheHitUUID: Option[UUID] = None) = {
  val (workflowRoot, missingErrors) = checkPreconditions(actual)
  // call to `diff`
}
def diffCheckTypes(actual: WorkflowFlatMetadata, workflowID: UUID, cacheHitUUID: Option[UUID] = None) = {
  val (workflowRoot, missingErrors) = checkPreconditions(actual)
  // call to `diffValuesTyped`
}

private def checkDiffPreconditions(actual: WorkflowFlatMetadata): (String, Set[String]) = {
  val workflowRoot: String = actual.value.get("workflowRoot").collectFirst { case JsString(r) => r } getOrElse "No Workflow Root"
  val missingErrors: Set[String] = value.keySet.diff(actual.value.keySet) map { k => s"Missing key: $k" }

  (workflowRoot, missingErrors)
}
```

Reply:
@aednichols, @cjllanwarne, @mcovarr on a second thought, I'm not sure, should we really introduce a test case (Centaur or unit) for just one specific combination of input parameters? Maybe we should just cover the specific cases (like semicolons in metadata keys) by fine-grained unit tests?

Running comparisons between maybe-not-yet-carbonited metadata and carbonited metadata in each Centaur test seems like a perfect way to make our long-running tests to run even longer 🙂

Reply:
my 2 cents: I'm okay with closing this for the reasons you state.